### DISCUSSION TITLE
How to improve performance of YouTube for VLESS-TCP-XTLS-Vision?

### DISCUSSION THREAD

**Discussion by iiMouse** [XTLS/Xray-core#3629] [Q&A]
Hello, I set up VLESS-TCP-XTLS-Vision. Speedtest.net download speed and ping seems fine, but YouTube videos are buffering very often at 1080p. I am the only user of the XRay proxy. How do I troubleshoot this situation, and are there ways to improve it?

**Comment by Any4Den** [comment]
Hi, you are not alone. I have the same problem. I have not tried on PC, but on mobile Android, even 1080p plays with brakes, periodically buffering. Although the speedtest via VPN gives 200 Mbit/s. I will follow the topic, maybe someone will tell me what the problem is.

upd: Low speed of YouTube via VPN in home network, via 4G it plays in 4K without freezing. Although in home network speedtest via VPN is 200 Mbit/s. And I need to watch YouTube on Android TV.

**Comment by Hisir0909** [comment]
Client block quic may work

**Comment by Any4Den** [comment]
In mobile browser it's working worst like through app. How can I blocking quick in client?

________________________________

**Reply by Hisir0909** [reply]
{
    "type": "field",
    "port": "443",
    "network": "udp",
    "outboundTag": "blocked"
}

this?
the better choose is blocking in home gateway

**Reply by Any4Den** [reply]
> { "type": "field", "port": "443", "network": "udp", "outboundTag": "blocked" }
> 
> this? the better choose is blocking in home gateway

Client v2ray, how can I set it up there?
Or do I need to change the vless client settings?
Or Xray? 3x-ui?
Sorry, I don't know much about this. I need a more detailed answer))

**Reply by Hisir0909** [reply]
> > { "type": "field", "port": "443", "network": "udp", "outboundTag": "blocked" }
> > this? the better choose is blocking in home gateway
> 
> Client v2ray, how can I set it up there? Or do I need to change the vless client settings? Or Xray? 3x-ui? Sorry, I don't know much about this. I need a more detailed answer))

If you are using a client like v2rayNG, you may need to customize the configuration file and block quic in the route rule. For information on how to write a configuration file, please refer to the XTLS documentation.

**Comment by TrajanBenn** [comment]
Try IPV6 solution, may be not suffering from a lot of restriction on bandwidth.

**Comment by laperuz92** [comment]
I can confirm disabling quic in 3x-ui inbound setting helped me with YT. Thanks to @Hisir0909 for the hint.
Here is the screenshot how to make it in 3x-ui
![Screenshot 2024-09-16 at 22 16 41](https://github.com/user-attachments/assets/ff437852-944f-4a10-86b2-95fbae90c112)

**Reply by vgdh** [reply]
confim, it helps

**Reply by Hellsingoff** [reply]
Confirm. With this option active, it stopped working a few days ago for some reason.

**Reply by turbcool** [reply]
Confirmed. My youtube on Android works flawlessly the moment I disabled QUIC.

**Reply by devopg** [reply]
We found a workaround, but maybe we shouldn’t ignore it and should report it as a bug to the developers?

**Reply by Sprucuis** [reply]
> We found a workaround, but maybe we shouldn’t ignore it and should report it as a bug to the developers?

I found that disabling Route only (but with all http / tls / quic / fakedns enabled) in my case solved the problem of long video loading when opened. Seems that there's a problem currently in Sniffing, especially when it comes to QUIC, but where exactly? Agreed, we need to report it to developer as it's an often bug as I can see to many forums topics (ntc.party if you ask)
